lib/rules/comment-pattern/__tests__/index.js

```diff
@@ -0,0 +1,113 @@
+'use strict';
+
+const { messages, ruleName } = require('..');
+
+testRule({
+	ruleName,
+	config: [/foo-.+/],
+
+	accept: [
+		{
+			code: '/* foo-valid yay */',
+		},
+		{
+			code: `/* foo--
+			multi-line
+			\n
+			\r
+			\r\n
+			\n\r
+			\t
+				*/`,
+		},
+		{
+			code: '.without-comment {}',
+		},
+	],
+
+	reject: [
+		{
+			code: '/* not foo- */',
+			message: messages.expected(/foo-.+/),
+		},
+		{
+			code: '/**/',
+			message: messages.expected(/foo-.+/),
+		},
+	],
+});
+
+testRule({
+	ruleName,
+	config: ['foo-.+'],
+
+	accept: [
+		{
+			code: '/* foo-valid yay */',
+		},
+		{
+			code: `/* foo--
+			multi-line
+			\n
+			\r
+			\r\n
+			\n\r
+			\t
+				*/`,
+		},
+		{
+			code: '.without-comment {}',
+		},
+	],
+
+	reject: [
+		{
+			code: '/* not foo- */',
+			message: messages.expected('foo-.+'),
+		},
+		{
+			code: '/**/',
+			message: messages.expected('foo-.+'),
+		},
+	],
+});
+
+testRule({
```

Comment:
I think we can remove this `testRule`.

---

I'd expect `config: [true]` to produce an invalid option error. This is likely a short coming in the `validateOptions` function, that we can pick up another time.

Reply:
I removed this. I'll open an issue soon 👍